presto-main/src/main/java/com/facebook/presto/sql/rewrite/MaterializedViewOptimizationRewriteUtils.java

```diff
@@ -42,12 +42,14 @@ public static Query optimizeQueryUsingMaterializedView(
         MaterializedViewCandidateExtractor materializedViewCandidateExtractor = new MaterializedViewCandidateExtractor(session, metadata);
         materializedViewCandidateExtractor.process(node);
         Set<QualifiedObjectName> materializedViewCandidates = materializedViewCandidateExtractor.getMaterializedViewCandidates();
-        if (materializedViewCandidates.isEmpty()) {
-            return node;
-        }
         // TODO: Select the most compatible and efficient materialized view for query rewrite optimization https://github.com/prestodb/presto/issues/16431
-        Query optimizedQuery = getQueryWithMaterializedViewOptimization(metadata, session, sqlParser, accessControl, node, materializedViewCandidates.iterator().next());
-        return optimizedQuery;
+        for (QualifiedObjectName candidate : materializedViewCandidates) {
```

Comment:
Lets leave the below issue as well as a TODO 
https://github.com/prestodb/presto/issues/16759